optaplanner-examples/src/main/java/org/optaplanner/examples/examination/domain/Examination.java

```diff
@@ -134,29 +143,4 @@ public void setScore(HardSoftScore score) {
         this.score = score;
     }
 
-    // ************************************************************************
-    // Complex methods
-    // ************************************************************************
-
-    @ProblemFactCollectionProperty
-    private List<TopicConflict> calculateTopicConflictList() {
-        List<TopicConflict> topicConflictList = new ArrayList<>();
-        for (Topic leftTopic : topicList) {
-            for (Topic rightTopic : topicList) {
-                if (leftTopic.getId() < rightTopic.getId()) {
-                    int studentSize = 0;
-                    for (Student student : leftTopic.getStudentList()) {
-                        if (rightTopic.getStudentList().contains(student)) {
-                            studentSize++;
-                        }
-                    }
-                    if (studentSize > 0) {
-                        topicConflictList.add(new TopicConflict(leftTopic, rightTopic, studentSize));
-                    }
-                }
-            }
-        }
-        return topicConflictList;
-    }
```

Comment:
As I was testing different examples, I found out that examination takes an ungodly amount of time to initialize. Often many many minutes. I fixed it - the student list is now a student set, because `contains()` is just incredibly painful on a list.

Reply:
I agree the studentList should be a studentSet (and yes a LinkedHashSet for reproducibility which this PR does)

Are we sure we want to make List<TopicConflict> part of the dataset generation (= increase the dataset XML file sizes) and not just calculate it on the fly once when the solver started like we did before?
I suspect your student List to Set change made this method a lot faster so it can stay here instead of going into the generator. Wdyt?